litmus-portal/backend/self-deployer/pkg/util/uitls.go

```diff
@@ -0,0 +1,101 @@
+package util
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/litmuschaos/litmus/litmus-portal/backend/self-deployer/pkg/k8s"
+
+)
+
+func WaitForServer(server string) {
+	log.Print("Checking if Server Up ", server)
+	for true {
+		req, err := http.NewRequest("GET", server, bytes.NewBuffer([]byte("")))
+		req.Header.Set("Content-Type", "application/json")
+		client := &http.Client{}
+		res, err := client.Do(req)
+		if err != nil || res.StatusCode != 200 {
+			time.Sleep(30 * time.Second)
+		} else {
+			break
+		}
+	}
+}
+
+func RegSelfCluster(server, pid string) (string, error) {
+	var jsonStr = []byte(`{"query":"mutation { userClusterReg(clusterInput: { cluster_name: \"Self-Cluster\", description:\"Self-Cluster\", project_id:\"` + pid + `\", platform_name:\"others\", cluster_type:\"INTERNAL\"})}"}`)
+	req, err := http.NewRequest("POST", server, bytes.NewBuffer(jsonStr))
+	if err != nil {
+		log.Print(err.Error())
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Print(err.Error())
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print(err.Error())
+		return "", err
+	}
+	regex := regexp.MustCompile(`"ey.*"`)
+	tk := regex.FindString(string(body))
+	tk = strings.Trim(tk, `"`)
+	return tk, nil
+}
+
+func Deploy(file string) error {
+	resp, err := http.Get(file)
+	if err != nil {
+		log.Print(err.Error())
+		return err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print(err.Error())
+		return err
+	}
+	log.Print(string(body))
+	response, err := k8s.ClusterOperations(body, "create")
+	if err != nil {
+		log.Println(err.Error())
+		return err
+	}
+	responseData, err := json.Marshal(response)
+	if err != nil {
+		log.Println("err:", err)
+	}
+	log.Print("SUBSCRIBER DEPLOYMENT RESPONSE : ", string(responseData))
+	return nil
+}
+
+func CleanUp(path string) error {
+	body, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Print(err.Error())
+		return err
+	}
+	response, err := k8s.ClusterOperations(body, "delete")
```

Comment:
(Same comment) reuse by importing from agent

Reply:
Same as above, cannot reuse unless tagged and published on github

Reply:
Same reason as above, my function takes different format of data. It might be a little extra work to convert it to your format. Maybe we can keep this as a more generic function as it takes an array of bytes and the request type, we can add a util function to convert from json in go-agent and use these function in there instead?